src/ui/hooks/settings.ts

```diff
@@ -179,53 +181,22 @@ function convertUserSettings(data: GetUserSettings | undefined): ExperimentalUse
     return emptySettings;
   }
 
-  const settings = data.viewer.settings;
+  const preferences = data.viewer.preferences;
   return {
     apiKeys: data.viewer.apiKeys as ApiKey[],
     defaultWorkspaceId: data.viewer.defaultWorkspace?.id || null,
-    disableLogRocket: settings.disableLogRocket,
-    enableTeams: settings.enableTeams,
-    role: settings.role || "developer",
+    disableLogRocket: preferences.disableLogRocket ?? false,
+    role: preferences.role ?? "developer",
```

Comment:
Since the backend doesn't know anything about the contents of `preferences`, the defaults are now defined in the frontend. This also implies a slight change in behavior: when we change the default for a preference that the user has never changed himself, he will be switched to the new default (because his `preferences` don't contain that preference). I like this better than the previous behavior: the settings were initialized in the backend at the time when the `user_settings` entry was created, if the defaults were changed by us later this would only apply to new users.

Reply:
Agreed. This matches the proposal in FE-1483 also.